internal/provider/kubernetes/gatewayclass.go

```diff
@@ -96,21 +102,27 @@ func (r *gatewayClassReconciler) Reconcile(ctx context.Context, request reconcil
 
 	var cc controlledClasses
 
+	found := false
 	for i := range gatewayClasses.Items {
 		if gatewayClasses.Items[i].Spec.ControllerName == r.controller {
 			cc.addMatch(&gatewayClasses.Items[i])
+			if gatewayClasses.Items[i].GetName() == request.Name {
+				found = true
+			}
 		}
 	}
+	if !found {
```

Comment:
should `r.resourceTable.GatewayClasses.Delete(request.Name)` be moved under `r.log.Info("failed to find gatewayclass", "name", request.Name)` ?